src/UITests/AIWinSession.cs

```diff
@@ -60,20 +60,44 @@ private void LaunchApplicationAndAttach()
             var configPathArgument = $"--ConfigFolder \"{Path.Combine(TestContext.TestResultsDirectory, TestContext.TestName)}\"";
 
             process = Process.Start(exePath, configPathArgument);
-            process.WaitForInputIdle();
 
-            // small buffer between splash screen disappearing 
-            // and main window initializing; otherwise in rare
-            // cases splash screen can be picked up as main window
-            Thread.Sleep(30000);
-
-            DesiredCapabilities appCapabilities = new DesiredCapabilities();
-            appCapabilities.SetCapability("appTopLevelWindow", process.MainWindowHandle.ToString("x"));
-            session = new WindowsDriver<WindowsElement>(new Uri(WindowsApplicationDriverUrl), appCapabilities);
+            const int attempts = 10; // this number should give us enough retries for the build to work
+            StartNewSessionWithRetry(attempts);
 
             driver = new AIWinDriver(session, process.Id);
         }
 
+        /// <summary>
+        /// We can't start a WinAppDriver session until ai-win is past its splash screen and fully loaded.
+        /// This takes a variable amount of time depending on the executing machine. It is particularly slow
+        /// in our build pipeline. Rather than set a long delay for the worst case scenario, we instead attempt
+        /// to start a new session repeatedly until ai-win is ready.
+        /// </summary>
+        /// <param name="attempts">Number of times to retry starting a new session</param>
+        private void StartNewSessionWithRetry(int attempts)
+        {
+            // if the session and its title are present, ai-win is ready for testing.
+            while (attempts > 0 && string.IsNullOrEmpty(session?.Title))
+            {
+                attempts--;
+
+                StartNewSession();
+                Thread.Sleep(3000);
+            }
+        }
+
+        private void StartNewSession()
+        {
+            try
+            {
+                process.Refresh(); // updates process.MainWindowHandle
```

Comment:
Food for thought: I tend to use the '_' char to connote private fields within a class. So, "_process" for example. This helps distinguish between local and member variables. I think it's especially useful when the member variable is used in several places throughout a class.

Reply:
Underscore adopted